src/hotspot/share/gc/g1/g1VMOperations.hpp

```diff
@@ -87,18 +87,34 @@ class VM_G1CollectForAllocation : public VM_CollectForAllocation {
 };
 
 // Concurrent G1 stop-the-world operations such as remark and cleanup.
-class VM_G1Concurrent : public VM_Operation {
-  VoidClosure* _cl;
-  const char*  _message;
+class VM_G1PauseConcurrent : public VM_Operation {
+private:
   uint         _gc_id;
+  const char*  _message;
+
+protected:
+  VM_G1PauseConcurrent(const char* message) :
+    _gc_id(GCId::current()), _message(message) { }
 
 public:
-  VM_G1Concurrent(VoidClosure* cl, const char* message) :
-    _cl(cl), _message(message), _gc_id(GCId::current()) { }
-  virtual VMOp_Type type() const { return VMOp_G1Concurrent; }
-  virtual void doit();
   virtual bool doit_prologue();
   virtual void doit_epilogue();
+  virtual void doit();
+  virtual void work() = 0;
```

Comment:
Please make `work()` protected - there does not seem to be a need to make it public.

Reply:
Done.